src/handler/recalls.ts

```diff
@@ -0,0 +1,89 @@
+import { getProfile } from '@dvsa/cvs-feature-flags/profiles/vtx';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { MotSecret } from '../models/motRecalls';
+import { formatErrorMessage } from '../util/errorMessage';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { filterMotRecalls, getBearerToken, getMotRecallsByVin } from '../util/recalls';
+import { validateSingleVin } from '../validators/recalls';
+
+const cache: Map<string, string> = new Map();
+const defaultResponse = addHttpHeaders({
+  statusCode: 200,
+  body: JSON.stringify({
+    manufacturer: null,
+    hasRecall: false,
+  }),
+});
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Recalls end point called');
+  try {
+    const featureFlags = await getProfile();
+
+    if (!featureFlags.recallsApi) {
+      logger.error('Recall Feature Flag is undefined');
+      return addHttpHeaders({
+        statusCode: 500,
+        body: 'Recall Feature Flag is undefined',
+      });
+    }
+
+    if (!featureFlags.recallsApi.enabled) {
+      logger.warn('Flag disabled: please enable for recalls functionality');
+      return defaultResponse;
+    }
+
+    const vin: string = decodeURIComponent(event.pathParameters?.vin as string);
+    if (!validateSingleVin(vin)) {
+      logger.error(formatErrorMessage('VIN provided in path parameter is not valid.'));
+      return defaultResponse;
+    }
+
+    const motSecret = {
+      clientID: process.env.CLIENT_ID,
+      clientSecret: process.env.CLIENT_SECRET,
+      scopeURL: process.env.SCOPE_URL,
+      accessTokenURL: process.env.ACCESS_TOKEN_URL,
+      apiKey: process.env.API_KEY,
+      apiURL: process.env.API_URL,
+    } as MotSecret;
+
+    if (!motSecret) {
```

Comment:
I'm not sure this code path will ever execute if we are creating the motSecret object above, it will always be truthy.